### PR TITLE
Add connect wallet button

### PR DESCRIPTION
**Pull request by radchukd**
I fixed wallet loading and added the "Connect a wallet" button along with some refactoring.

### CODE REVIEW

**Review by apbendi — COMMENTED** [review]
Hey @radchukd, looking good! Nice work.

One thing I did notice is clicking on my address after connecting, which used to allow the user to change wallets, now produces an error:

Screen Shot 2021-09-07 at 7 28 21 AM

Screen Shot 2021-09-07 at 7 28 35 AM

File: frontend/src/store/wallet.ts
Comment (mds1) [inline frontend/src/store/wallet.ts:163]: A nit, but would prefer a more meaningful parameter name than `l`, maybe `status`?

File: frontend/src/store/wallet.ts
Comment (mds1) [inline frontend/src/store/wallet.ts:288]: Why do we need this new casting?

Comment (mds1) [reply frontend/src/store/wallet.ts:288]: Maybe better to do `} catch (error as {code: string}) {`

File: frontend/src/layouts/BaseLayout.vue
Comment (mds1) [inline frontend/src/layouts/BaseLayout.vue:25]: My localStorage has `__q_strn|MetaMask` as the value for the `last-wallet` key, but when testing out this PR it gets stuck with `isLoading = true`, and there's a few (possibly related) errors in the console. Some screenshots below (might need to open the image in a new tab to see the full resolution)

image

image

image

Comment (radchukd) [reply frontend/src/layouts/BaseLayout.vue:25]: What exactly did you do?

File: frontend/src/store/wallet.ts
Comment (radchukd) [reply frontend/src/store/wallet.ts:288]: New casting is needed because of `Object is of type 'unknown'.ts(2571)`

`error as {code: string}` This doesn't work because of `Catch clause variable type annotation must be 'any' or 'unknown' if specified.ts(1196)` https://github.com/microsoft/TypeScript/issues/20024.

I fixed this casting like this:
```
 catch (switchError) {
      const { code } = switchError as { code: number };
      if (code === 4902) {}
}
```

File: frontend/src/layouts/BaseLayout.vue
Comment (mds1) [reply frontend/src/layouts/BaseLayout.vue:25]: I just built the app locally like normal. Also just noticed it seems my screenshot of the header was poorly timed as its from the small portion of the spinner cycle when there is no spinner 🙃

> Try clearing site data.

I thought about this, but didn't want to do it, and don't think we should merge this PR if it requires clearing site data—the reason is because if you need to clear site data for it to work properly, then the live app will break for users once deployed and we don't want to make them clear their data

Comment (radchukd) [reply frontend/src/layouts/BaseLayout.vue:25]: Yep, you're right. I think I got it now.

Comment (mds1) [reply frontend/src/layouts/BaseLayout.vue:25]: Yes, seems to be working now, thank you!

**Review by mds1 — APPROVED** [review]
LGTM, thanks @radchukd!

@apbendi I'll wait for your final review before merging

**Pull request merged**
